### DISCUSSION TITLE
How to tag multiple objects in an s3 bucket using aws cli?

### DISCUSSION THREAD

**Discussion by ajmalicf** [aws/aws-cli#7273] [Q&A] [answered]
Is there a way to apply a tag to multiple objects in an S3 bucket using one single put-object-tagging cli command?
i.e. if I have two files (file-1.ext, file-2.ext) I need to run the following two commands:

`aws s3api delete-object-tagging --bucket bucket-name --key file-1.ext   --tagging 'TagSet=[{Key=key1,Value=value1}]'`
`aws s3api delete-object-tagging --bucket bucket-name --key file-2.ext   --tagging 'TagSet=[{Key=key1,Value=value1}]'`

**Comment by aBurmeseDev** [comment, marked as answer]
Hi @ajmalicf, thank you for reaching out. Unfortunately, AWS CLI commands for S3 API are only meant to work with one single key value and in this case, you might have to run the command a few times for multiple objects. Here's docs for your reference: https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3api/delete-object-tagging.html

Hope it helps!

**Reply by aBurmeseDev** [reply]
Although here's an alternative approach I would suggest checking out: https://docs.aws.amazon.com/AmazonS3/latest/userguide/batch-ops-delete-object-tagging.html
